### DISCUSSION TITLE
Debugging when using "var body: some ReducerOf<Self>"

### DISCUSSION THREAD

**Discussion by rzolin** [pointfreeco/swift-composable-architecture#2476] [Q&A]
Could somebody help me out with debugging TCA when I use reducer structure like below:
```
    public var body: some ReducerOf<Self> {
        
        Reduce<State, Action> { state, action in
            switch action {
             case ...
            }
       }
       
   }
```

I would like to stick `.debug()` somewhere here to see what actions are flying around. But get errors.
Maybe there is a simple way to convert it into the `var reducer =` which would work with `.debug()`

**Comment by Armenm** [comment]
I use `_printChanges()` instead. Have you tried below?
```swift
public var body: some ReducerProtocolOf<Self> {
    Reduce<State, Action> { _, action in
        switch action {
        case .closeButtonTapped:
            return .none
        }
    }
    ._printChanges()
}
```

**Comment by rzolin** [comment]
Thanks @Armenm, that worked beautifully 👍
